src/vs/base/browser/markdownRenderer.ts

```diff
@@ -536,17 +536,20 @@ export function renderStringAsPlaintext(string: IMarkdownString | string) {
 
 /**
  * Strips all markdown from `markdown`. For example `# Header` would be output as `Header`.
- * provide @param withCodeBlocks to retain code blocks
+ * provide @param accessibleView to return the markdown for the accessible view, which retains code blocks and doesn't sanitize the output
  */
-export function renderMarkdownAsPlaintext(markdown: IMarkdownString, withCodeBlocks?: boolean) {
+export function renderMarkdownAsPlaintext(markdown: IMarkdownString, accessibleView?: boolean) {
 	// values that are too long will freeze the UI
 	let value = markdown.value ?? '';
 	if (value.length > 100_000) {
 		value = `${value.substr(0, 100_000)}…`;
 	}
 
-	const html = marked.parse(value, { async: false, renderer: withCodeBlocks ? plainTextWithCodeBlocksRenderer.value : plainTextRenderer.value }).replace(/&(#\d+|[a-zA-Z]+);/g, m => unescapeInfo.get(m) ?? m);
+	const html = marked.parse(value, { async: false, renderer: accessibleView ? plainTextWithCodeBlocksRenderer.value : plainTextRenderer.value }).replace(/&(#\d+|[a-zA-Z]+);/g, m => unescapeInfo.get(m) ?? m);
 
+	if (accessibleView) {
```

Comment:
We should still sanitize. We can't rely on the markdown rendering to do this for us

What issues was the sanitization causing?

Reply:
it's replacing -> with &lt

Reply:
Where is this content used?

Reply:
in the accessible view (see the linked bug)

Reply:
<img width="712" alt="Screenshot 2024-09-09 at 10 24 23 AM" src="https://github.com/user-attachments/assets/b3d6fe0c-28ed-4a19-bf4f-a513358abb65">

Reply:
Can you encode characters like `<` before sanitizing? Or decode them after sanitizing? Or see if there's a way to stop the sanitizer encoding them

Bypassing the sanitizer seems like a very blunt instrument to solve a small seeming issue

Reply:
Sanitization should ensure any bad html that slips through the plaintext rendering gets stripped out. That should also mean that the content the accessibility view shows more closely matches what will actually be rendered in chat

Reply:
Unclear why this isn't happening in the chat response 🤔